crates/revm/src/database.rs

```diff
@@ -10,6 +10,7 @@ use revm::{
 pub type SubState<DB> = CacheDB<State<DB>>;
 
 /// Wrapper around StateProvider that implements revm database trait
+#[derive(Debug, Clone)]
 pub struct State<DB: StateProvider>(pub DB);
```

Comment:
Ah nice - this should be generally useful for fast revm simulations directly on the db.